packaging_automation/package_cloud_statistics_collector.py

```diff
@@ -0,0 +1,164 @@
+import argparse
+import json
+import time
+from datetime import datetime, date
+from enum import Enum
+from http import HTTPStatus
+from typing import List, Any
+
+import requests
+from sqlalchemy import Column, INTEGER, DATE, TIMESTAMP, String
+
+from .common_tool_methods import (remove_suffix, stat_get_request)
+from .dbconfig import (Base, db_session, DbParams, RequestType)
+
+ORGANIZATION_NAME = "citusdata"
+PC_PACKAGE_COUNT_SUFFIX = " packages"
+PC_DOWNLOAD_DATE_FORMAT = '%Y%m%dZ'
+
+
+class PackageCloudRepos(Enum):
+    community = "community"
+    enterprise = "enterprise"
+    azure = "azure"
+    community_nightlies = "community-nightlies"
+    enterprise_nightlies = "enterprise-nightlies"
+    test = "test"
+
+
+class PackageCloudOrganizations(Enum):
+    citusdata = "citusdata"
+    citus_bot = "citus-bot"
+
+
+class PackageCloudDownloadStats(Base):
+    __tablename__ = "package_cloud_download_stats"
+    id = Column(INTEGER, primary_key=True, autoincrement=True)
+    fetch_date = Column(TIMESTAMP, nullable=False)
+    package_name = Column(String, nullable=False)
+    package_full_name = Column(String, nullable=False)
+    package_version = Column(String, nullable=False)
+    package_release = Column(String, nullable=False)
+    package_type = Column(String, nullable=False)
+    download_date = Column(DATE, nullable=False)
+    download_count = Column(INTEGER, nullable=False)
+
+
+def package_count(organization: PackageCloudOrganizations, repo_name: PackageCloudRepos,
+                  package_cloud_api_token: str) -> int:
+    result = requests.get(
+        f"https://{package_cloud_api_token}:@packagecloud.io/api/v1/repos.json?include_collaborations=true")
+
+    repo_details = json.loads(result.content)
+    for repo in repo_details:
+        if repo["fqname"] == f"{organization.name}/{repo_name.name}":
+            return int(remove_suffix(repo['package_count_human'], PC_PACKAGE_COUNT_SUFFIX))
+    raise ValueError(f"Repo name with the name {repo_name} could not be found on package cloud")
```

Comment:
Is this supposed to be `repo_name.name`?

Reply:
Fixed